### DISCUSSION TITLE
how to pass displayObjects from one container to another and gracefully remove the emptied container?

### DISCUSSION THREAD

**Discussion by kayden1940** [pixijs/pixijs#10192] [Q&A] [answered]
when I try to dispose an empty container those previously contained sprites would also get removed, even though it's no longer under the container it looks something like this:
```
  for (let c1i = 0; c1i < c1.children.length; c1i++) {
    const c1s = c1.children[c1i];
    c1s.setParent(c2);
  }
  c1.destroy();

```
what happen is c1.destroy() removes the previously added sprites inside c1,
same thing happens using app.stage.removechild(c1)
https://codesandbox.io/p/devbox/v8lzq9
here is an example, I was expecting the sprites to not get removed
thanks

**Comment by gabrielliwerant** [comment, marked as answer]
I think this is happening because objects in js are passed by reference, meaning that it doesn't matter where you move them, the references are maintained, hence if the container is set to destroy its object references when it is destroyed, it will seek out and find those sprite objects no matter where they are.

Therefore, I don't think there's actually a graceful way to move sprites between containers. I think the best you can do is to "clone" the sprite, meaning copying over all the important features of the old sprite into a new one and then deleting the old one. Pixi doesn't have any built-in way to clone as of v7, so you'll have to make a utility for it yourself.

There's also a plugin called pixi layers, which might be worth a look, as it claims to be able to further abstract the notion of sprite ordering despite their containers, and hence might solve whatever it is you're trying to do, although I don't have personal experience using pixi layers.
